bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/threading/ThreadsafeTimers.java

```diff
@@ -55,19 +60,60 @@ public void setIdentifier(String identifier) {
     }
 
     /**
-     * Schedules a callback to run at a given time.
+     * Schedules a block of code for later execution.
      *
-     * @param id timerId to append to the identifier base for naming the scheduled job
-     * @param zdt time to schedule the job
-     * @param callback function to run at the given time
-     * @return a {@link ScheduledCompletableFuture}
+     * @param instant the point in time when the code should be executed
+     * @param closure the code block to execute
+     * @return a handle to the created timer, so that it can be canceled or rescheduled
      */
-    private ScheduledCompletableFuture<Object> createFuture(long id, ZonedDateTime zdt, Runnable callback) {
-        return scheduler.schedule(() -> {
+    public Timer createTimer(ZonedDateTime instant, Runnable closure) {
+        return createTimer(identifier, instant, closure);
+    }
+
+    /**
+     * Schedules a block of code for later execution.
+     *
+     * @param identifier an optional identifier
+     * @param instant the point in time when the code should be executed
+     * @param closure the code block to execute
+     * @return a handle to the created timer, so that it can be canceled or rescheduled
+     */
+    public Timer createTimer(@Nullable String identifier, ZonedDateTime instant, Runnable closure) {
+        return scriptExecution.createTimer(identifier, instant, () -> {
             synchronized (lock) {
-                callback.run();
+                closure.run();
             }
-        }, identifier + ".timeout." + id, zdt.toInstant());
+        });
+    }
+
+    /**
+     * Schedules a block of code (with argument) for later execution
+     *
+     * @param instant the point in time when the code should be executed
+     * @param arg1 the argument to pass to the code block
+     * @param closure the code block to execute
+     * @return a handle to the created timer, so that it can be canceled or rescheduled
+     */
+    public Timer createTimerWithArgument(ZonedDateTime instant, Object arg1, Consumer<Object> closure) {
```

Comment:
I have two questions about this method. 

1) Presumably your test should actually be:

```
ThreadsafeTimers.createTimerWithArgument(time.toZDT(), arg1, (newArg) => {
    console.info(`Hello world from createTimerWithArgument without identifier and argument: ${newArg}`);
});
```

2) What is the use case here? Why not allow the closure to capture the state? This is effectively what your original test actually did, as the passed parameter was unused. Which is why I'm not sure I understand the use-case.

Reply:
> Presumably your test should actually be:

Yes, thanks.

> What is the use case here? Why not allow the closure to capture the state?

TBH, I am not sure what the use case is here. I do not use this in any of my scripts.

Reply:
In that case I would suggest that we remove the `createTimeWithArgument` methods. I _suspect_ that the reason they are here is because:

- The original Java class included this method because it's not so simple to include state when calling from Java, so passing state in (whilst providing the same callback interface) is very convenient
- These were by default just extended to scripts

I don't really see the need in JS because people will pass closures that capture the state implicitly. Having these extra methods just mean extra code to maintain, plus a more confusing API for script writers.

Reply:
Good idea, I also don't see the need. 

> plus a more confusing API for script writers.

That is a really good argumen.